packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -544,6 +554,7 @@ abstract class ScrollPosition extends ViewportOffset with ScrollMetrics {
       _didChangeViewportDimensionOrReceiveCorrection = false;
       _pendingDimensions = true;
       if (haveDimensions && !correctForNewDimensions(_lastMetrics!, currentMetrics!)) {
+        _maybeDispatchScrollMetricsNotification();
```

Comment:
Why not let `_RenderSingleChildViewport.performLayout` response the return value like what `RenderViewport.performLayout` does? 
It should do multi times otherwise the ScrollPosition has a wrong state, such as the `_lastMetrics` does not updated.